src/app/shared/form/builder/ds-dynamic-form-ui/models/autocomplete/ds-dynamic-autocomplete.component.ts

```diff
@@ -1,18 +1,53 @@
 import { FormGroup } from '@angular/forms';
-import { DynamicTagModel } from '../tag/dynamic-tag.model';
 import { NgbTypeahead, NgbTypeaheadSelectItemEvent } from '@ng-bootstrap/ng-bootstrap';
-import { Observable, of as observableOf } from 'rxjs';
+import { BehaviorSubject, Observable, of as observableOf } from 'rxjs';
 import { PageInfo } from '../../../../../../core/shared/page-info.model';
 import { VocabularyService } from '../../../../../../core/submission/vocabularies/vocabulary.service';
 import { DynamicFormLayoutService, DynamicFormValidationService } from '@ng-dynamic-forms/core';
 import { catchError, debounceTime, distinctUntilChanged, map, merge, switchMap, tap } from 'rxjs/operators';
-import { buildPaginatedList } from '../../../../../../core/data/paginated-list.model';
-import { isEmpty, isNotEmpty } from '../../../../../empty.util';
+import { buildPaginatedList, PaginatedList } from '../../../../../../core/data/paginated-list.model';
+import { isEmpty, isNotEmpty, isNull } from '../../../../../empty.util';
 import { DsDynamicTagComponent } from '../tag/dynamic-tag.component';
 import { MetadataValueDataService } from '../../../../../../core/data/metadata-value-data.service';
 import { FormFieldMetadataValueObject } from '../../../models/form-field-metadata-value.model';
 import { LookupRelationService } from '../../../../../../core/data/lookup-relation.service';
+import {
+  AUTOCOMPLETE_CUSTOM_JSON_PREFIX,
+  AUTOCOMPLETE_CUSTOM_SOLR_PREFIX,
+  DsDynamicAutocompleteModel
+} from './ds-dynamic-autocomplete.model';
+import { getFirstSucceededRemoteDataPayload } from '../../../../../../core/shared/operators';
+import { GetRequest } from '../../../../../../core/data/request.models';
+import { HttpOptions } from '../../../../../../core/dspace-rest/dspace-rest.service';
+import { HttpParams } from '@angular/common/http';
+import { RequestService } from '../../../../../../core/data/request.service';
+import { RemoteDataBuildService } from '../../../../../../core/cache/builders/remote-data-build.service';
+import { HALEndpointService } from '../../../../../../core/shared/hal-endpoint.service';
+import { RemoteData } from '../../../../../../core/data/remote-data';
+import { VocabularyEntry } from '../../../../../../core/submission/vocabularies/models/vocabulary-entry.model';
+import { ConfigurationDataService } from '../../../../../../core/data/configuration-data.service';
+import { CANONICAL_PREFIX_KEY } from '../../../../../handle.service';
+import { ConfigurationProperty } from '../../../../../../core/shared/configuration-property.model';
+import { DsDynamicAutocompleteService } from './ds-dynamic-autocomplete.service';
+import { TranslateService } from '@ngx-translate/core';
+
+/**
+ * Prefix for custom autocomplete definition from the `submission-forms.xml`.
+ * <input-type autocomplete-custom="solr-handle_title_ac">autocomplete</input-type>
+ */
+const AUTOCOMPLETE_CUSTOM_HANDLE_TITLE = 'solr-handle_title_ac';
```

Comment:
@milanmajchrak TT about how it works, can we specify field we want autocomplate from from UI? unsafe

Reply:
`AUTOCOMPLETE_CUSTOM_HANDLE_TITLE` is used only for prettifying the suggestion.